src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java

```diff
@@ -88,23 +83,63 @@ public ConfigDef config() {
   @Override
   public Config validate(Map<String, String> connectorConfigs) {
     Config config = super.validate(connectorConfigs);
```

Comment:
We can't instantiate `DatagenConnectorConfig` here since the C3 UI also calls validate to generate the connector creation form at which point none of the configs are set (including mandatory ones like `kafka.topic`).

`getConfigValue` follows the pattern established [here](https://www.confluent.io/blog/write-a-kafka-connect-connector-with-configuration-handling/)

Reply:
In case any of the individual validations fail, instantiating `DatagenConnectorConfig` fails with a `ConfigException`. This can also be used as a check to avoid any further validations.

Reply:
If the config doesn't have a valid schema source, are there any validations that would still be worth doing? We wouldn't be able to do key field validation, for example.

Reply:
Actually, the only validation that would be skipped is the schema key field validation which makes complete sense to skip in that scenario anyway. I've refactored the code accordingly